lib/commands/deprecate.js

```diff
@@ -56,17 +57,26 @@ class Deprecate extends BaseCommand {
     const versions = Object.keys(packument.versions)
       .filter(v => semver.satisfies(v, spec, { includePrerelease: true }))
 
+    const dryRun = this.npm.config.get('dry-run')
+
     if (versions.length) {
       for (const v of versions) {
         packument.versions[v].deprecated = msg
+        if (msg) {
+          log.notice(`deprecating ${packument.name}@${v} with message "${msg}"`)
+        } else {
+          log.notice(`undeprecating ${packument.name}@${v}`)
+        }
+      }
+      if (!dryRun) {
+        return otplease(this.npm, this.npm.flatOptions, opts => npmFetch(uri, {
+          ...opts,
+          spec: p,
+          method: 'PUT',
+          body: packument,
```

Comment:
Aside: I'd like to get away from the client sending the entire packument, including versions that aren't being modified.

Publish, in contrast, only sends the new version being published: https://github.com/npm/cli/blob/0a71ebb3d1e7fca07d39ea64e5f1acd22ecd255c/workspaces/libnpmpublish/lib/publish.js#L89-L98